src/cascadia/TerminalSettingsEditor/ActionsViewModel.idl

```diff
@@ -1,60 +1,188 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 
+import "EnumEntry.idl";
+import "ColorSchemeViewModel.idl";
+import "MainPage.idl";
+
 namespace Microsoft.Terminal.Settings.Editor
 {
-    runtimeclass ModifyKeyBindingEventArgs
+    [default_interface] runtimeclass ArgsTemplateSelectors : Windows.UI.Xaml.Controls.DataTemplateSelector
+    {
+        ArgsTemplateSelectors();
+
+        Windows.UI.Xaml.DataTemplate Int32Template;
+        Windows.UI.Xaml.DataTemplate Int32OptionalTemplate;
+        Windows.UI.Xaml.DataTemplate UInt32Template;
+        Windows.UI.Xaml.DataTemplate UInt32OptionalTemplate;
+        Windows.UI.Xaml.DataTemplate FloatTemplate;
+        Windows.UI.Xaml.DataTemplate StringTemplate;
+        Windows.UI.Xaml.DataTemplate ColorSchemeTemplate;
+        Windows.UI.Xaml.DataTemplate FilePickerTemplate;
+        Windows.UI.Xaml.DataTemplate FolderPickerTemplate;
+        Windows.UI.Xaml.DataTemplate BoolTemplate;
+        Windows.UI.Xaml.DataTemplate BoolOptionalTemplate;
+        Windows.UI.Xaml.DataTemplate EnumTemplate;
+        Windows.UI.Xaml.DataTemplate FlagTemplate;
+        Windows.UI.Xaml.DataTemplate TerminalCoreColorOptionalTemplate;
+        Windows.UI.Xaml.DataTemplate WindowsUIColorOptionalTemplate;
+    }
+
+    [default_interface] runtimeclass EditAction : Windows.UI.Xaml.Controls.Page, Windows.UI.Xaml.Data.INotifyPropertyChanged
+    {
+        EditAction();
+        CommandViewModel ViewModel { get; };
+    }
+
+    runtimeclass NavigateToCommandArgs
+    {
+        CommandViewModel Command { get; };
+        IHostedInWindow WindowRoot { get; };
+    }
+
+    runtimeclass ModifyKeyChordEventArgs
     {
         Microsoft.Terminal.Control.KeyChord OldKeys { get; };
         Microsoft.Terminal.Control.KeyChord NewKeys { get; };
-        String OldActionName { get; };
-        String NewActionName { get; };
     }
 
-    runtimeclass KeyBindingViewModel : Windows.UI.Xaml.Data.INotifyPropertyChanged
+    runtimeclass CommandViewModel : Windows.UI.Xaml.Data.INotifyPropertyChanged
     {
         // Settings Model side
+        String Name;
+        String ID { get; };
+        Boolean IsUserAction { get; };
+        // keybindings
+        IObservableVector<KeyChordViewModel> KeyChordViewModelList { get; };
```

Comment:
I wasn't going to say "This should be called `KeyChordList`" until I got to the bottom and saw that you _do_ have a `CommandList` (which is a list of CommandViewModel).

nit: I'd recommend making the names similar in style (and I would drop `ViewModel` from the member name to do that)